### ISSUE TITLE
ROUND() evaluates to -nan

### ISSUE THREAD

**Issue by mrigger** [cwida/duckdb#521]
Consider the following statement:

```sql
SELECT ROUND(0.1, 1000); -- -nan
```
The SELECT results in `-nan`, which might be unexpected and could lead to surprising results. For example, neither of the following fetches a row, and it seems to be impossible to check whether a certain value is a NaN value:

```sql
CREATE TABLE t0(c0 INT);
INSERT INTO t0(c0) VALUES (0);
SELECT * FROM t0 WHERE t0.c0 > ROUND(0.1, 1000); -- {}
SELECT * FROM t0 WHERE t0.c0 <= ROUND(0.1, 1000); -- {}
```
I found this based on the latest master commit (c52fc9bafd22e41a5554d3305530ebc1bf8364f9), and also verified that the latest fix branch is affected (215e40cd5cdb7eee68797df4982219c63f04e2a9).

**Comment by Mytherin** [comment]
Nice find. Indeed round should never result in isnan.

**Comment by Mytherin** [comment]
Fixed in e399d40eb76f2fbe97c9e139551171862ab7848d. The problem was that round was computed as follows:
```python
double modifier = pow(10, precision);
return (round(input * modifier)) / modifier;
```

A large precision lead to `pow` returning NaN. I replaced this implementation with an incremental computation of the `modifier` to ensure that NaN can never be returned; a too high precision will just be truncated to the highest possible precision that does not result in NaN.
